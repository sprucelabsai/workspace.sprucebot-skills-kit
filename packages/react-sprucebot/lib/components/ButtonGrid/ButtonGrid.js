"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GridButton = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var GridButton =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(GridButton, _Component);

  function GridButton() {
    (0, _classCallCheck2.default)(this, GridButton);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(GridButton).apply(this, arguments));
  }

  (0, _createClass2.default)(GridButton, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          selected = _this$props.selected,
          children = _this$props.children,
          subtitle = _this$props.subtitle,
          rest = (0, _objectWithoutProperties2.default)(_this$props, ["className", "selected", "children", "subtitle"]);
      return _react.default.createElement("a", (0, _extends2.default)({
        className: "btn ".concat(className || '', " ").concat(selected ? 'selected' : '')
      }, rest), _react.default.createElement("span", null, children, subtitle && _react.default.createElement("br", null), subtitle && _react.default.createElement("small", null, subtitle)));
    }
  }]);
  return GridButton;
}(_react.Component);

exports.GridButton = GridButton;
GridButton.propTypes = {
  selected: _propTypes.default.bool,
  subtitle: _propTypes.default.string
};
GridButton.defaultProps = {
  selected: false
};