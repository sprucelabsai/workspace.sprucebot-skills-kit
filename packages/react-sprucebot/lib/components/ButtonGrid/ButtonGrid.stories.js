'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonActions = require('@storybook/addon-actions');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _ButtonGrid = require('./ButtonGrid');

var _ButtonGrid2 = require('./ButtonGrid.md');

var _ButtonGrid3 = _interopRequireDefault(_ButtonGrid2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('ButtonGrid', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_ButtonGrid3.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		'div',
		{ className: 'single_col' },
		_react2.default.createElement(
			_ButtonGrid.ButtonGrid,
			null,
			_react2.default.createElement(
				_ButtonGrid.GridButton,
				{
					selected: (0, _react4.boolean)('Select Option 1', false),
					subtitle: 'It\'s good',
					onClick: (0, _addonActions.action)('Select tile')
				},
				'Option 1'
			),
			_react2.default.createElement(
				_ButtonGrid.GridButton,
				{
					selected: (0, _react4.boolean)('Select Option 2', false),
					subtitle: 'It\'s better!',
					onClick: (0, _addonActions.action)('Select tile')
				},
				'Option 2'
			),
			_react2.default.createElement(
				_ButtonGrid.GridButton,
				{
					selected: (0, _react4.boolean)('Select Option 3', true),
					subtitle: 'Def worth a look',
					onClick: (0, _addonActions.action)('Select tile')
				},
				'Option 3'
			)
		),
		_react2.default.createElement(
			_ButtonGrid.ButtonGrid,
			null,
			_react2.default.createElement(
				_ButtonGrid.GridButton,
				{
					selected: (0, _react4.boolean)('Select ðŸ¤–', true),
					onClick: (0, _addonActions.action)('Select tile')
				},
				'\uD83E\uDD16'
			),
			_react2.default.createElement(
				_ButtonGrid.GridButton,
				{
					selected: (0, _react4.boolean)('Select ðŸ”¥ #1', false),
					onClick: (0, _addonActions.action)('Select tile')
				},
				'\uD83D\uDD25'
			),
			_react2.default.createElement(
				_ButtonGrid.GridButton,
				{
					selected: (0, _react4.boolean)('Select ðŸ”¥ #2', false),
					onClick: (0, _addonActions.action)('Select tile')
				},
				'\uD83D\uDD25'
			),
			_react2.default.createElement(
				_ButtonGrid.GridButton,
				{
					selected: (0, _react4.boolean)('Select ðŸŒ²', true),
					onClick: (0, _addonActions.action)('Select tile')
				},
				'\uD83C\uDF32'
			)
		)
	);
})));