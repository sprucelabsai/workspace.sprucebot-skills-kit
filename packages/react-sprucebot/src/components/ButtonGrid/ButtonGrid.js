import styled from 'styled-components'
import React, { Component } from 'react'
import PropTypes from 'prop-types'

export const ButtonGrid = styled.div.attrs({
	className: 'btn_grid'
})``

export class GridButton extends Component {
	render() {
		const props = Object.assign({}, this.props)
		const { className, selected, children, subtitle, ...rest } = props

		return (
			<a
				className={`btn ${className || ''} ${selected ? 'selected' : ''}`}
				{...rest}
			>
				<span>
					{children}
					{subtitle && <br />}
					{subtitle && <small>{subtitle}</small>}
				</span>
			</a>
		)
	}
}

GridButton.propTypes = {
	selected: PropTypes.bool,
	subtitle: PropTypes.string
}

GridButton.defaultProps = {
	selected: false
}
