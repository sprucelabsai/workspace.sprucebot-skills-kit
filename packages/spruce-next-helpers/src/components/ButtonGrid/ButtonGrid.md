# ButtonGrid

Set the line-height of .btn_grid .btn to control height of button.
The text will stay centered.
