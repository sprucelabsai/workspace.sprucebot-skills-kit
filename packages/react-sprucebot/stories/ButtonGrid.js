import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withInfo } from '@storybook/addon-info'
import { withKnobs, boolean, select } from '@storybook/addon-knobs/react'
import { withReadme, withDocs } from 'storybook-readme'
import { GridButton, ButtonGrid } from '../lib/components/ButtonGrid/ButtonGrid'
import readme from './docs/ButtonGrid.md'

const stories = storiesOf('ButtonGrid', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(
		readme,
		withInfo()(() => (
			<div className="single_col">
				<ButtonGrid>
					<GridButton
						selected={boolean('Select Option 1', false)}
						subtitle="It's good"
						onClick={action('Select tile')}
					>
						Option 1
					</GridButton>
					<GridButton
						selected={boolean('Select Option 2', false)}
						subtitle="It's better!"
						onClick={action('Select tile')}
					>
						Option 2
					</GridButton>
					<GridButton
						selected={boolean('Select Option 3', true)}
						subtitle="Def worth a look"
						onClick={action('Select tile')}
					>
						Option 3
					</GridButton>
				</ButtonGrid>

				<ButtonGrid>
					<GridButton
						selected={boolean('Select ðŸ¤–', true)}
						onClick={action('Select tile')}
					>
						ðŸ¤–
					</GridButton>
					<GridButton
						selected={boolean('Select ðŸ”¥ #1', false)}
						onClick={action('Select tile')}
					>
						ðŸ”¥
					</GridButton>
					<GridButton
						selected={boolean('Select ðŸ”¥ #2', false)}
						onClick={action('Select tile')}
					>
						ðŸ”¥
					</GridButton>
					<GridButton
						selected={boolean('Select ðŸŒ²', true)}
						onClick={action('Select tile')}
					>
						ðŸŒ²
					</GridButton>
				</ButtonGrid>
			</div>
		))
	)
)
