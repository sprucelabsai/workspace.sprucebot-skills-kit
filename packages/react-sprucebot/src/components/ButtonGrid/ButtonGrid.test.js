import React from 'react'
import ButtonGrid from './ButtonGrid'
import { shallow } from 'enzyme'

describe('ButtonGrid tests', () => {
	it('Should match the snapshot', () => {
		const renderedComponent = shallow(<ButtonGrid />)

		expect(renderedComponent).toMatchSnapshot()
	})
})
