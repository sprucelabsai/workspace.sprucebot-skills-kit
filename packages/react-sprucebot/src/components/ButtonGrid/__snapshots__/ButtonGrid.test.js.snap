// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGrid renders 1`] = `
<div
  className="ButtonGrid-s120cl1n-0 btn_grid bCkfqO"
/>
`;

exports[`ButtonGrid renders a GridButton 1`] = `
<div
  className="ButtonGrid-s120cl1n-0 btn_grid bCkfqO"
>
  <a
    className="btn  "
  >
    <span />
  </a>
</div>
`;

exports[`GridButton renders 1`] = `
<a
  className="btn  "
>
  <span />
</a>
`;
